### PR TITLE
[WIP] - Port ember component tests to new test harness

### PR DESCRIPTION
**Pull request by scalvert**
As per #561

### CODE REVIEW

File: packages/@glimmer/runtime/test/yield-test.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    this.render({
+      layout: 'In layout -- {{yield}}',
+      template: 'In template',
+    });
+
+    this.assertComponent('In layout -- In template');
+    this.assertStableRerender();
+  }
+
+  /*
```
Comment (scalvert) [inline packages/@glimmer/runtime/test/yield-test.ts]: Any thoughts on a good way to translate these data driven tests into class-based tests, @chadhietala?

Comment (chadhietala) [inline packages/@glimmer/runtime/test/yield-test.ts]: I would just unroll that loop.

Comment (chadhietala) [inline packages/@glimmer/runtime/test/yield-test.ts]: You could also move the loop in a test case and call the test `primitive type asserts`

Comment (scalvert) [inline packages/@glimmer/runtime/test/yield-test.ts]: OK - I'm inclined to go for the unroll approach, as it's more explicit what we're testing when we do that.

**Comment by chadhietala** [comment]
Part of #614

**Pull request closed without merging**